src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -830,6 +830,8 @@ private TypeWithAnnotations BindVariableTypeWithAnnotations(CSharpSyntaxNode dec
                     // Keep processing it as a non-const local.
                     isConst = false;
                 }
+
+                declType.Type.CheckRecordLangVersion(diagnostics, typeSyntax, typeSyntax.Location);
```

Comment:
>declType.Type.CheckRecordLangVersion(diagnostics, typeSyntax, typeSyntax.Location); [](start = 16, length = 83)

Where are we making sure that we are not inside a method body? It looks like ```record x;``` inside a body is never parsed as a record declaration.